lib/Dialect/FIRRTL/Transforms/Lint.cpp

```diff
@@ -0,0 +1,58 @@
+//===- Lint.cpp -------------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "mlir/IR/Threading.h"
+#include "llvm/ADT/APSInt.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace firrtl;
+
+namespace {
+struct LintPass : public LintBase<LintPass> {
+  void runOnOperation() override {
+    auto fModule = getOperation();
+    (void)failableParallelForEach(
+        fModule.getContext(), fModule.getOps(),
```

Comment:
I'm not sure if parallel over operations is a good idea? Thread-to-work ratio seems a bit rough, but dunno.

Misc potential preconditions/assumptions:

1) post ExpandWhen's / single-block 
2) IMDCE already executed (in theory if this was run before such, what if the assert is not reachable/dead?).

That sound about right? Just trying to understand this and it's role/purpose/scope.

And pipeline-ordering-wise probably of course want to run this after various optimizations such that if our compiler can/will prove these are statically false/lint-worthy, we'll have done so (e.g., IMCP).

Reply:
You are right, parallel work seems very small. Removed it.
It is expected that all the optimizations are done, additionally post expand-whens, hence running it late in the pipeline.
I will update the comments to indicate the assumptions.